### PR TITLE
Adds functionality to provision or add an instance

### PR DESCRIPTION
**Pull request by AlexSCorey**
##### SUMMARY
This adds functionality to provision/add an mesh instance.  You have to have a kubernetes deployment, or override the IS_K8S variable [here](https://github.com/ansible/awx/compare/feature-mesh-scaling...AlexSCorey:6036-t-AddInstance?expand=1#diff-828a318d5bf2548b3b7ebc88148b7cf48ae20fe6062afea8ddcef439f20c9938R48)

##### ISSUE TYPE
 - New or Enhanced Feature

##### COMPONENT NAME
 - UI

### CODE REVIEW

File: awx/ui/src/screens/Instances/InstanceList/InstanceList.js [hunk 1]
```diff
@@ -135,6 +140,15 @@ function InstanceList() {
                 onExpandAll={expandAll}
                 qsConfig={QS_CONFIG}
                 additionalControls={[
+                  ...(isK8
```
Comment (kialam) [inline awx/ui/src/screens/Instances/InstanceList/InstanceList.js]: we should also add a check for if the logged in user is a SysAdmin

File: awx/ui/src/screens/Instances/Shared/InstanceForm.js [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import {
+  ExecutionEnvironmentLookup,
+  InstanceGroupsLookup,
+} from 'components/Lookup';
+
+// This is hard coded because the API does not have the ability to send us a list that contains
+// only the types of instances that can be added.  Control and Hybrid instances cannot be added.
+
+const INSTANCE_TYPES = [
+  { id: 2, name: t`Execution`, value: 'execution' },
```
Comment (kialam) [inline awx/ui/src/screens/Instances/Shared/InstanceForm.js]: `id` here is used by `AnsibleSelect` to set the `select` element id attribute. We're not passing it as part of the payload and should make this a string value and not an integer.

File: awx/ui/src/screens/Instances/InstanceAdd/InstanceAdd.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    const { instanceGroups, executionEnvironment } = values;
+    values.execution_environment = executionEnvironment?.id;
+
+    try {
+      const {
+        data: { id },
+      } = await InstancesAPI.create(values);
+
+      for (const group of instanceGroups) {
+        await InstancesAPI.associateInstanceGroup(id, group.id);
```
Comment (kialam) [inline awx/ui/src/screens/Instances/InstanceAdd/InstanceAdd.js]: spoke with @jbradberry and I think we want to allow default behavior here. E.g. the user should be able to create an Instance and not associate it with any instance group. They can always associate it later in the InstancesGroup list page.

File: awx/ui/src/screens/Instances/Shared/InstanceForm.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+            label: type.name,
+            isDisabled: false,
+          }))}
+          value={instanceType.value}
+          onChange={(e, opt) => {
+            instanceTypeHelpers.setValue(opt);
+          }}
+        />
+      </FormGroup>
+      <InstanceGroupsLookup
```
Comment (kialam) [inline awx/ui/src/screens/Instances/Shared/InstanceForm.js]: Same as the comment earlier, we probably don't need this lookup.

File: awx/ui/src/api/models/Instances.js [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import Base from '../Base';
+import InstanceGroupsMixin from '../mixins/InstanceGroups.mixin';
 
-class Instances extends Base {
+class Instances extends InstanceGroupsMixin(Base) {
```
Comment (kialam) [inline awx/ui/src/api/models/Instances.js]: Not sure we need this.

File: awx/ui/src/screens/Instances/InstanceList/InstanceList.js (hunk 1 above)
Comment (AlexSCorey) [reply awx/ui/src/screens/Instances/InstanceList/InstanceList.js:143]: I thought about that...and I'm happy to do that, but this section of the app is only available to sys admins

File: awx/ui/src/screens/Instances/InstanceList/InstanceList.js [hunk 6]
```diff
@@ -135,6 +142,15 @@ function InstanceList() {
                 onExpandAll={expandAll}
                 qsConfig={QS_CONFIG}
                 additionalControls={[
+                  ...(isK8 && me.is_superuser
```
Comment (jbradberry) [inline awx/ui/src/screens/Instances/InstanceList/InstanceList.js:145]: minor terminology pick: K8S is short for K + 8 characters (ubernete) + S.  You should probably keep the S.

File: awx/ui/src/screens/Instances/InstanceAdd/InstanceAdd.test.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { act } from 'react-dom/test-utils';
+import { createMemoryHistory } from 'history';
+import { InstancesAPI } from 'api';
+import {
+  mountWithContexts,
```
Comment (kialam) [inline awx/ui/src/screens/Instances/InstanceAdd/InstanceAdd.test.js:6]: For net new unit tests we should be using react testing library and not enzyme, however we can merge the PR for now with a follow up ticket to convert later.

File: awx/ui/src/screens/Instances/InstanceList/InstanceList.js (hunk 6 above)
Comment (kialam) [inline awx/ui/src/screens/Instances/InstanceList/InstanceList.js:145]: <3

**Review by kialam — APPROVED** [review]

**Pull request merged**